backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +63,128 @@ private void updatePersonFacility(Person person, UUID facilityId) {
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), false);
+  // Specifications filters for queries
+  private Specification<Person> inWholeOrganizationFilter() {
+    return (root, query, cb) ->
+        cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId());
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getAllPatients(int pageOffset, int pageSize) {
-    return _repo
-        .findAllByOrganization(
-            _os.getCurrentOrganization(), false, PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // Note: Patients with NULL facilityIds appear in ALL facilities.
+  private Specification<Person> inFacilityFilter(@NotNull UUID facilityId) {
+    return (root, query, cb) ->
+        cb.and(
+            cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId()),
+            cb.or(
+                cb.isNull(root.get(SpecField.Facility)), // null check first
+                cb.equal(root.get(SpecField.Facility).get(SpecField.InternalId), facilityId)));
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getAllPatientsCount() {
-    return _repo.countAllByOrganization(_os.getCurrentOrganization(), false);
+  private Specification<Person> nameMatchesFilter(
+      @Size(min = MINIMUMCHARFORSEARCH) String namePrefixMatch) {
+    String likeString = namePrefixMatch.trim().toLowerCase() + "%";
+    return (root, query, cb) ->
+        cb.or(
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.FirstName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.MiddleName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.LastName)), likeString));
   }
 
-  // FYI archived is not just a parameter because it requires different permissions.
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public List<Person> getArchivedPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            true,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  private Specification<Person> isDeletedFilter(boolean isDeleted) {
+    return (root, query, cb) -> cb.equal(root.get(SpecField.IsDeleted), isDeleted);
+  }
+
+  // called by List function and Count function
+  private boolean checkPermissionsForListFunc(
+      UUID facilityId, boolean isArchived, String searchTerm) {
+    if (_auth.userHasSiteAdminRole()) { // site admins skip all other checks
+      return true;
+    }
+
+    ArrayList<UserPermission> perms = new ArrayList<UserPermission>();
+
+    if (facilityId == null) {
+      // READ_PATIENT_LIST does NOT seem right, probably an admin role to see all patients across
```

Comment:
this will be UserPermission.ACCESS_ALL_FACILITIES soon, make a comment and i'll add it in. for right now, this if-condition should be a no-op.